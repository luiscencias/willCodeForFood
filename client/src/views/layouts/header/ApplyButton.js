import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';

import { Link } from 'react-router-dom';

const StyledButton = withStyles(theme => ({
	root: {
		boxShadow: 'none',
		color: theme.palette.common.white,
		marginLeft: theme.spacing.unit / 2,

		'&:hover': {
			backgroundColor: '#00bcd4'
		},

		[ theme.breakpoints.down('xs') ]: {
			padding: '10px 8px',
			minWidth: '96px'
		}
	}
}))(Button);

const ApplyButton = () => (
	<StyledButton component={Link} to="/apply" variant="contained" color="secondary" size="small">
		Apply
	</StyledButton>
);

export default ApplyButton;