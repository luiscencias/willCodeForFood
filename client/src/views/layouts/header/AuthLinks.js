import React from 'react';

import { AuthLink } from './styles';

import { Link } from 'react-router-dom';

const AuthLinks = () => (
	<>
		<AuthLink
			variant="outlined"
			color="secondary"
			size="small"
		>
			Log In
		</AuthLink>
		<AuthLink
			variant="contained"
			color="secondary"
			size="small"
		>
			Apply
		</AuthLink>
	</>
);

export default AuthLinks;