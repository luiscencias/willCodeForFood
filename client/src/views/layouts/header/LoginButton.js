import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';

const StyledButton = withStyles(theme => ({
	root: {
		[ theme.breakpoints.down('xs') ]: {
			margin: theme.spacing.unit * 2,
			padding: `11px ${ theme.spacing.unit / 2 }px`,
			borderRadius: theme.spacing.unit
		}
	}
}))(Button);

const LoginButton = ({ onClick }) => (
	<StyledButton variant="outlined" color="secondary" size="small" onClick={onClick}>
		Sign In
	</StyledButton>
);

LoginButton.propTypes = {
	onClick: PropTypes.func
}

export default LoginButton;