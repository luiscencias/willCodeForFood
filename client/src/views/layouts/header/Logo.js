import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';

import { Link } from 'react-router-dom';

export const StyledButton = withStyles(theme => ({
	label: {
		fontWeight: 900
	}
}))(Button);

const Logo = () => <StyledButton size="small" component={Link} to="/" inline >PBE</StyledButton>;

export default Logo;