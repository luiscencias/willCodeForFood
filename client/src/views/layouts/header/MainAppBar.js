import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';


const StyledAppBar = withStyles(theme => ({
	root: {
		boxShadow: 'none'
	}
}))(AppBar);


const StyledToolBar = withStyles(theme => ({
	root: {
		height: theme.spacing.mainHeader.height
	}
}))(Toolbar);


const MainAppBar = ({ children }) => (
	<StyledAppBar position="sticky" color="default">
		<StyledToolBar>
			{ children }
		</StyledToolBar>
	</StyledAppBar>
);

MainAppBar.propTypes = {
	children: PropTypes.node.isRequired
};

export default MainAppBar;