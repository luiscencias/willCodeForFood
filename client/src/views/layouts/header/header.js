import React, { useEffect, useRef, useState } from 'react';

import Hidden from '@material-ui/core/Hidden';

import MainHeader from './MainHeader';

import DesktopNav from './DesktopNav';
import MobileNav from './MobileNav';

const routes = [
	{name: "Home", path: '/', exact: true},
	{name: 'Dashboard', path: '/dashboard', authRequired: true},
	{name: "About Us", path: '/about'},
	{name: 'Membership', path: '/membership'},
	{name: "Events", path: '/events'},
	{name: "Shop", path: '/shop'}
];

const createHandleScroll = setScrolled => () => {
	setScrolled( window.scrollY > 0 );
}

const Header = ({ isAuthenticated = false }) => {
	const [ scrolled, setScrolled ] = useState(false);

	useEffect(() => {
		const handleScroll = createHandleScroll(setScrolled);

		window.addEventListener("scroll", handleScroll);

		// Will be run when header unmounts.
		return () => window.removeEventListener("scroll", handleScroll);
	}, []);


	return (
		<MainHeader scrolled={scrolled}>
			<Hidden smDown>
				<DesktopNav 
					routes={ routes } 
					isAuthenticated={isAuthenticated}
					scrolled={scrolled}
				/>
			</Hidden>
			<Hidden mdUp>
				<MobileNav 
					routes={ routes } 
					isAuthenticated={isAuthenticated} 
					scrolled={scrolled}
				/>
			</Hidden>
		</MainHeader>
	);
};

export default Header;