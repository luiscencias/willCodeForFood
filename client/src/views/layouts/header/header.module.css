.navLink: {
	font-weight: 200;
	color: red;
}