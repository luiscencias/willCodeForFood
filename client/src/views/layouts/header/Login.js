import React, { useState } from 'react';
import PropTypes from 'prop-types';

import LoginButton from './LoginButton';

const Login = () => {
	const [ modalIsVisible, setModalVisibility ] = useState( false );

	return <LoginButton onClick={() => setModalVisibility( true )} />;
};

export default Login;