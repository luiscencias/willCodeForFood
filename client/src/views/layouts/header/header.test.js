import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, mount } from 'enzyme';
import { navLinks } from './header'
import {Button} from "@material-ui/core";

describe("navLinks", () => {

    it('renders', () => {

        const button = shallow(<Button/>);
        expect(button.exists()).toBe(true);

    });

    it('has right names for links', () => {
        expect(navLinks[0]['name']).toEqual('About');
        expect(navLinks[1]['name']).toEqual('Events');
        expect(navLinks[2]['name']).toEqual('Members');
        expect(navLinks[3]['name']).toEqual('Gallery');
        expect(navLinks[4]['name']).toEqual('Shop');
    });

});


