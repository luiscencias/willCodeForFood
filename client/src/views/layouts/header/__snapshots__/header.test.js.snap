// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without changing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Hidden
          implementation="js"
          lgDown={false}
          lgUp={false}
          mdDown={false}
          mdUp={false}
          smDown={true}
          smUp={false}
          xlDown={false}
          xlUp={false}
          xsDown={false}
          xsUp={false}
        >
          <DesktopNav
            isAuthenticated={false}
            routes={
              Array [
                Object {
                  "exact": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "authRequired": true,
                  "name": "Dashboard",
                  "path": "/dashboard",
                },
                Object {
                  "name": "About Us",
                  "path": "/about",
                },
                Object {
                  "name": "Membership",
                  "path": "/membership",
                },
                Object {
                  "name": "Events",
                  "path": "/events",
                },
                Object {
                  "name": "Shop",
                  "path": "/shop",
                },
              ]
            }
          />
        </Hidden>,
        <Hidden
          implementation="js"
          lgDown={false}
          lgUp={false}
          mdDown={false}
          mdUp={true}
          smDown={false}
          smUp={false}
          xlDown={false}
          xlUp={false}
          xsDown={false}
          xsUp={false}
        >
          <MobileNav
            isAuthenticated={false}
            routes={
              Array [
                Object {
                  "exact": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "authRequired": true,
                  "name": "Dashboard",
                  "path": "/dashboard",
                },
                Object {
                  "name": "About Us",
                  "path": "/about",
                },
                Object {
                  "name": "Membership",
                  "path": "/membership",
                },
                Object {
                  "name": "Events",
                  "path": "/events",
                },
                Object {
                  "name": "Shop",
                  "path": "/shop",
                },
              ]
            }
          />
        </Hidden>,
      ],
      "scrolled": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <DesktopNav
            isAuthenticated={false}
            routes={
              Array [
                Object {
                  "exact": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "authRequired": true,
                  "name": "Dashboard",
                  "path": "/dashboard",
                },
                Object {
                  "name": "About Us",
                  "path": "/about",
                },
                Object {
                  "name": "Membership",
                  "path": "/membership",
                },
                Object {
                  "name": "Events",
                  "path": "/events",
                },
                Object {
                  "name": "Shop",
                  "path": "/shop",
                },
              ]
            }
          />,
          "implementation": "js",
          "lgDown": false,
          "lgUp": false,
          "mdDown": false,
          "mdUp": false,
          "smDown": true,
          "smUp": false,
          "xlDown": false,
          "xlUp": false,
          "xsDown": false,
          "xsUp": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isAuthenticated": false,
            "routes": Array [
              Object {
                "exact": true,
                "name": "Home",
                "path": "/",
              },
              Object {
                "authRequired": true,
                "name": "Dashboard",
                "path": "/dashboard",
              },
              Object {
                "name": "About Us",
                "path": "/about",
              },
              Object {
                "name": "Membership",
                "path": "/membership",
              },
              Object {
                "name": "Events",
                "path": "/events",
              },
              Object {
                "name": "Shop",
                "path": "/shop",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <MobileNav
            isAuthenticated={false}
            routes={
              Array [
                Object {
                  "exact": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "authRequired": true,
                  "name": "Dashboard",
                  "path": "/dashboard",
                },
                Object {
                  "name": "About Us",
                  "path": "/about",
                },
                Object {
                  "name": "Membership",
                  "path": "/membership",
                },
                Object {
                  "name": "Events",
                  "path": "/events",
                },
                Object {
                  "name": "Shop",
                  "path": "/shop",
                },
              ]
            }
          />,
          "implementation": "js",
          "lgDown": false,
          "lgUp": false,
          "mdDown": false,
          "mdUp": true,
          "smDown": false,
          "smUp": false,
          "xlDown": false,
          "xlUp": false,
          "xsDown": false,
          "xsUp": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "isAuthenticated": false,
            "routes": Array [
              Object {
                "exact": true,
                "name": "Home",
                "path": "/",
              },
              Object {
                "authRequired": true,
                "name": "Dashboard",
                "path": "/dashboard",
              },
              Object {
                "name": "About Us",
                "path": "/about",
              },
              Object {
                "name": "Membership",
                "path": "/membership",
              },
              Object {
                "name": "Events",
                "path": "/events",
              },
              Object {
                "name": "Shop",
                "path": "/shop",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Hidden
            implementation="js"
            lgDown={false}
            lgUp={false}
            mdDown={false}
            mdUp={false}
            smDown={true}
            smUp={false}
            xlDown={false}
            xlUp={false}
            xsDown={false}
            xsUp={false}
          >
            <DesktopNav
              isAuthenticated={false}
              routes={
                Array [
                  Object {
                    "exact": true,
                    "name": "Home",
                    "path": "/",
                  },
                  Object {
                    "authRequired": true,
                    "name": "Dashboard",
                    "path": "/dashboard",
                  },
                  Object {
                    "name": "About Us",
                    "path": "/about",
                  },
                  Object {
                    "name": "Membership",
                    "path": "/membership",
                  },
                  Object {
                    "name": "Events",
                    "path": "/events",
                  },
                  Object {
                    "name": "Shop",
                    "path": "/shop",
                  },
                ]
              }
            />
          </Hidden>,
          <Hidden
            implementation="js"
            lgDown={false}
            lgUp={false}
            mdDown={false}
            mdUp={true}
            smDown={false}
            smUp={false}
            xlDown={false}
            xlUp={false}
            xsDown={false}
            xsUp={false}
          >
            <MobileNav
              isAuthenticated={false}
              routes={
                Array [
                  Object {
                    "exact": true,
                    "name": "Home",
                    "path": "/",
                  },
                  Object {
                    "authRequired": true,
                    "name": "Dashboard",
                    "path": "/dashboard",
                  },
                  Object {
                    "name": "About Us",
                    "path": "/about",
                  },
                  Object {
                    "name": "Membership",
                    "path": "/membership",
                  },
                  Object {
                    "name": "Events",
                    "path": "/events",
                  },
                  Object {
                    "name": "Shop",
                    "path": "/shop",
                  },
                ]
              }
            />
          </Hidden>,
        ],
        "scrolled": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <DesktopNav
              isAuthenticated={false}
              routes={
                Array [
                  Object {
                    "exact": true,
                    "name": "Home",
                    "path": "/",
                  },
                  Object {
                    "authRequired": true,
                    "name": "Dashboard",
                    "path": "/dashboard",
                  },
                  Object {
                    "name": "About Us",
                    "path": "/about",
                  },
                  Object {
                    "name": "Membership",
                    "path": "/membership",
                  },
                  Object {
                    "name": "Events",
                    "path": "/events",
                  },
                  Object {
                    "name": "Shop",
                    "path": "/shop",
                  },
                ]
              }
            />,
            "implementation": "js",
            "lgDown": false,
            "lgUp": false,
            "mdDown": false,
            "mdUp": false,
            "smDown": true,
            "smUp": false,
            "xlDown": false,
            "xlUp": false,
            "xsDown": false,
            "xsUp": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isAuthenticated": false,
              "routes": Array [
                Object {
                  "exact": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "authRequired": true,
                  "name": "Dashboard",
                  "path": "/dashboard",
                },
                Object {
                  "name": "About Us",
                  "path": "/about",
                },
                Object {
                  "name": "Membership",
                  "path": "/membership",
                },
                Object {
                  "name": "Events",
                  "path": "/events",
                },
                Object {
                  "name": "Shop",
                  "path": "/shop",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <MobileNav
              isAuthenticated={false}
              routes={
                Array [
                  Object {
                    "exact": true,
                    "name": "Home",
                    "path": "/",
                  },
                  Object {
                    "authRequired": true,
                    "name": "Dashboard",
                    "path": "/dashboard",
                  },
                  Object {
                    "name": "About Us",
                    "path": "/about",
                  },
                  Object {
                    "name": "Membership",
                    "path": "/membership",
                  },
                  Object {
                    "name": "Events",
                    "path": "/events",
                  },
                  Object {
                    "name": "Shop",
                    "path": "/shop",
                  },
                ]
              }
            />,
            "implementation": "js",
            "lgDown": false,
            "lgUp": false,
            "mdDown": false,
            "mdUp": true,
            "smDown": false,
            "smUp": false,
            "xlDown": false,
            "xlUp": false,
            "xsDown": false,
            "xsUp": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "isAuthenticated": false,
              "routes": Array [
                Object {
                  "exact": true,
                  "name": "Home",
                  "path": "/",
                },
                Object {
                  "authRequired": true,
                  "name": "Dashboard",
                  "path": "/dashboard",
                },
                Object {
                  "name": "About Us",
                  "path": "/about",
                },
                Object {
                  "name": "Membership",
                  "path": "/membership",
                },
                Object {
                  "name": "Events",
                  "path": "/events",
                },
                Object {
                  "name": "Shop",
                  "path": "/shop",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
